qlib/strategy/base.py

```diff
@@ -154,10 +165,14 @@ def alter_outer_trade_decision(self, outer_trade_decision: BaseTradeDecision):
         ----------
         outer_trade_decision : BaseTradeDecision
             the decision updated by the outer strategy
+
+        Returns
+        -------
+            BaseTradeDecision
         """
         # default to reset the decision directly
         # NOTE: normally, user should do something to the strategy due to the change of outer decision
-        raise NotImplementedError(f"Please implement the `alter_outer_trade_decision` method")
+        # raise NotImplementedError(f"Please implement the `alter_outer_trade_decision` method")
```

Comment:
Why do you comment it out?

Reply:
Reverted this change. I personally do not like unimplemented abstract methods, but it should be OK to leave it for now.